samcli/commands/test_runner/init/cli.py

```diff
@@ -0,0 +1,204 @@
+SUPPORTED_RUNTIMES = ["python3.8"]
+
+LOG = logging.getLogger(__name__)
+COLOR = Colored()
+
+
+@click.command("init", help=HELP_TEXT, short_help=SHORT_HELP)
+@click.option(
+    "--tag-key",
+    type=str,
+    required=False,
+    help="""
+Specify the key of the tag used to discover resources.
+
+Discovered resources will have IAM statement templates generated within the Test Runner CloudFormation Template,
+and a mapping between that resource name and its ARN included in the environment variable specification file.
+""",
+)
+@click.option(
+    "--tag-value",
+    type=str,
+    required=False,
+    help="""
+Specify the value of the tag used to discover resources.
+
+Discovered resources will have IAM statement templates generated within the Test Runner CloudFormation Template,
+and a mapping between that resource name and its ARN included in the environment variable specification file.
+""",
+)
+@click.option(
+    "--template-name",
+    required=False,
+    type=str,
+    help="Specify the name of the generated Test Runner CloudFormation template.",
+    default=DEFAULT_TEST_RUNNER_CFN_TEMPLATE_FILE_NAME,
+)
+@click.option(
+    "--env-file",
+    required=False,
+    type=str,
+    help="""
+Specify the name of the generated resource-ARN map YAML file. This file can be passed to `sam test-runner run`, exposing the mappings as environment variables to the Fargate container.
+""",
+    default=DEFAULT_RESOURCE_ARN_MAP_FILE_NAME,
+)
+@click.option(
+    "--runtime",
+    required=False,
+    type=click.Choice(SUPPORTED_RUNTIMES, case_sensitive=False),
+    help=f"Specify the runtime with which to run tests.",
+    default="python3.8",
+)
+@click.option(
+    "--image-uri",
+    required=True,
+    type=str,
+    help="THIS IS A TEMPORARY OPTION. Supply an image URI for Fargate to create a container to run tests.",
+)
+@pass_context
+def cli(
+    ctx: Any,
+    tag_key: Optional[str],
+    tag_value: Optional[str],
+    template_name: str,
+    env_file: str,
+    runtime: str,
+    image_uri: str,
+) -> None:
+    """
+    `sam test-runner init` command entry point
+    """
+
+    do_cli(
+        ctx=ctx,
+        tag_key=tag_key,
+        tag_value=tag_value,
+        template_name=template_name,
+        env_file=env_file,
+        runtime=runtime,
+        image_uri=image_uri,
+    )
+
+
+def do_cli(
+    ctx: Any,
+    tag_key: Optional[str],
+    tag_value: Optional[str],
+    template_name: str,
+    env_file: str,
+    runtime: str,
+    image_uri: str,
+) -> None:
+    """
+    implementation of the `sam test-runner init` command
+    """
+
+    # TODO: Merge these two options into one to follow AWS CLI (if we can get access to their parser)
+    if tag_key and not tag_value:
+        LOG.info(COLOR.red("Specified tag key without a tag value."))
+        return
+
+    if tag_value and not tag_key:
+        LOG.info(COLOR.red("Specified tag value without a tag key."))
+        return
+
+    if tag_value and tag_key:
```

Comment:
Similar to the below comment on line 149. I would do `tag_filters = None` and then assign it with this if statement.